packages/hash/api/src/graph/knowledge/primitive/link-entity.ts

```diff
@@ -0,0 +1,159 @@
+
+export type CreateLinkEntityParams = {
+  ownedById: OwnedById;
+  properties?: PropertyObject;
+  linkEntityType: EntityTypeWithMetadata;
+  leftEntityId: EntityId;
+  leftToRightOrder?: number;
+  rightEntityId: EntityId;
+  rightToLeftOrder?: number;
+  actorId: AccountId;
+};
+
+export type LinkEntity = Entity & {
+  linkData: NonNullable<Entity["linkData"]>;
+};
+
+export const isEntityLinkEntity = (entity: Entity): entity is LinkEntity =>
+  !!entity.linkData;
+
+/**
+ * Create a link entity between a left and a right entity.
+ *
+ * @param params.ownedById - the id of the account who owns the new link entity
+ * @param params.linkEntityType - the link entity type of the link entity
+ * @param params.leftEntityId - the ID of the left entity
+ * @param params.leftToRightOrder (optional) - the left to right order of the link entity
+ * @param params.rightEntityId - the ID of the right entity
+ * @param params.rightToLeftOrder (optional) - the right to left order of the link entity
+ * @param params.actorId - the id of the account that is creating the link
+ */
+export const createLinkEntity: ImpureGraphFunction<
+  CreateLinkEntityParams,
+  Promise<LinkEntity>
+> = async ({ graphApi }, params) => {
+  const {
+    ownedById,
+    linkEntityType,
+    actorId,
+    leftEntityId,
+    leftToRightOrder,
+    rightEntityId,
+    rightToLeftOrder,
+    properties = {},
+  } = params;
+
+  if (!isEntityTypeLinkEntityType({ entityType: linkEntityType })) {
+    throw new Error(
+      `Entity type with ID "${linkEntityType.schema.$id}" is not a link entity type.`,
+    );
+  }
+
+  const linkData: LinkData = {
+    leftEntityId,
+    leftToRightOrder,
+    rightEntityId,
+    rightToLeftOrder,
+  };
+
+  const { data: linkEntityMetadata } = await graphApi.createEntity({
+    ownedById,
+    linkData,
+    actorId,
+    entityTypeId: linkEntityType.schema.$id,
+    properties,
+  });
+
+  return {
+    metadata: linkEntityMetadata as EntityMetadata,
+    properties,
+    linkData,
+  };
+};
+
+/**
+ * Update a link entity.
+ *
+ * @param params.linkEntity - the link entity being updated
+ * @param params.properties (optional) - the updated properties object of the link entity
+ * @param params.leftToRightOrder (optional) - the updated left to right order of the link entity
+ * @param params.rightToLeftOrder (optional) - the updated right to left order of the link entity
+ * @param params.actorId - the id of the account that is updating the entity
+ */
+export const updateLinkEntity: ImpureGraphFunction<
+  {
+    linkEntity: LinkEntity;
+    properties?: PropertyObject;
+    leftToRightOrder?: number;
+    rightToLeftOrder?: number;
+    actorId: AccountId;
+  },
+  Promise<LinkEntity>
+> = async ({ graphApi }, params) => {
+  const { actorId, leftToRightOrder, rightToLeftOrder, linkEntity } = params;
+
+  const properties = params.properties ?? linkEntity.properties;
+
+  const { data: metadata } = await graphApi.updateEntity({
```

Comment:
I realized this as I was talking with Tim. We need to explicitly supply an `archived` value here with the changes he has made

Reply:
Addressed in https://github.com/hashintel/hash/pull/1681/commits/aebc7fa2eb697046c7517b8e3fd1be91b6ed2f8b